### DISCUSSION TITLE
How enable the MQTT-Support with Docker-Compose

### DISCUSSION THREAD

**Discussion by grojas123** [microcks/microcks#1017] [Q&A] [answered]
I am checking this:
https://microcks.io/documentation/guides/mqtt-support/
I installed with: https://microcks.io/documentation/installing/docker-compose/
How enable the MQTT-Support with Docker-Compose?

**Comment by lbroudoux** [comment, marked as answer]
Hello!

To enable MQTT from docker-compose, you'll need to tweak the `config/application.properties` file under the docker-compose file.  This file contains properties coming from https://github.com/microcks/microcks/blob/master/minions/async/src/main/resources/application.properties that may be prefixed with `%docker-compose` that is the name of the profile used for launching from docker-compose.

So basically, you'll have to uncomment or add the following block with correct values:

```properties
%docker-compose.mqtt.server=my-mqtt-server:1883
%docker-compose.mqtt.username=microcks
%docker-compose.mqtt.password=microcks
```

You'll also need to add MQTT to the list of supported protocols using this property line:

```properties
# Explicitly telling the minion the protocols we want to support
%docker-compose.minion.supported-bindings=KAFKA,WS,MQTT
```

Finally, if you want the UI to display the correct MQTT server address, you'll also need to edit the `config/features.properties` file, changing this line:

```properties
features.feature.async-api.endpoint-MQTT=my-mqtt-server:1883
```

Hope it helps!

**Reply by grojas123** [reply]
Thank you
